shop/src/main/java/ro/msg/learning/shop/model/BaseEntity.java

```diff
@@ -0,0 +1,34 @@
+package ro.msg.learning.shop.model;
+
+
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.MappedSuperclass;
+
+@MappedSuperclass
+@Getter
+@Setter
+@ToString
+@RequiredArgsConstructor
+@SuperBuilder
+public abstract class BaseEntity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
You can use @EqualsAndHashCode annotation from Lombok instead of adding manually these methods.